### PR TITLE
MX-13883: native auth server use api for fetching the block

### PR DESCRIPTION
**Pull request by dragos-rebegea**

### CODE REVIEW

File: authentication/native/server_test.go [hunk 1]
```diff
@@ -81,35 +94,39 @@ func TestNativeserver_Validate(t *testing.T) {
...
-	hyperblockTimestamp := int64(10)
+	blockTimestamp := int64(10)
 	t.Run("proxy returns error should error", func(t *testing.T) {
 		t.Parallel()
 
 		args := createMockArgsNativeAuthServer()
-		args.Proxy = &testsCommon.ProxyStub{
-			GetHyperBlockByHashCalled: func(ctx context.Context, hash string) (*data.HyperBlock, error) {
-				return nil, expectedErr
+		args.HttpClientWrapper = &testsCommon.HTTPClientWrapperStub{
```
Comment (sstanculeanu) [inline authentication/native/server_test.go:102]: should also update test name

File: authentication/native/server.go [hunk 2]
```diff
@@ -117,3 +126,17 @@ func (server *authServer) validateSignature(token authentication.AuthToken) erro
 func (server *authServer) IsInterfaceNil() bool {
 	return server == nil
 }
+
+func (server *authServer) getBlockByHash(ctx context.Context, hash string) (*data.Block, error) {
+	var block data.Block
+	buff, code, err := server.httpClientWrapper.GetHTTP(ctx, server.apiNetworkAddress+"/blocks/"+hash)
```
Comment (sstanculeanu) [inline authentication/native/server.go:132]: extract "/blocks/" in a constant then the full path concatenation in a local var

File: data/block.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package data
+
+type Block struct {
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply data/block.go:3]: I think we can define only the used fields by the native auth.
I think timestamp is the only one we need? as the hash we already give on the API call and we don't check it again.

Comment (sstanculeanu) [inline data/block.go:3]: missing comment on exported

File: authentication/native/server.go [hunk 4]
```diff
@@ -2,41 +2,49 @@ package native
...
 
 type authServer struct {
-	proxy           workflows.ProxyHandler
-	tokenHandler    authentication.AuthTokenHandler
-	signer          builders.Signer
-	keyGenerator    crypto.KeyGenerator
-	pubKeyConverter core.PubkeyConverter
-	getTimeHandler  func() time.Time
+	httpClientWrapper authentication.HttpClientWrapper
+	apiNetworkAddress string
```
Comment (sstanculeanu) [inline authentication/native/server.go:29]: not used anymore

File: authentication/native/server.go (hunk 2 above)
Comment (dragos-rebegea) [reply authentication/native/server.go]: done

File: authentication/native/server_test.go (hunk 1 above)
Comment (dragos-rebegea) [reply authentication/native/server_test.go:102]: updated

File: data/block.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dragos-rebegea) [reply data/block.go:3]: done

File: authentication/native/server.go (hunk 4 above)
Comment (dragos-rebegea) [reply authentication/native/server.go]: removed

**Review by sstanculeanu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**